File: src/main/java/io/github/cdgeass/editor/dom/element/mapper/Parameter.java
package io.github.cdgeass.editor.dom.element.mapper;

import com.intellij.util.xml.Attribute;
import com.intellij.util.xml.Convert;
import com.intellij.util.xml.DomElement;
import com.intellij.util.xml.GenericAttributeValue;
import io.github.cdgeass.editor.dom.element.convert.ResultMapReferenceConvert;

/**
 * @author cdgeass
 * @since  2020-06-20
 */
public interface Parameter extends DomElement {

    GenericAttributeValue<String> getProperty();

    GenericAttributeValue<String> getJavaType();

    GenericAttributeValue<String> getJdbcType();

    GenericAttributeValue<String> getMode();

    @Attribute("resultMap")
    @Convert(ResultMapReferenceConvert.class)
    GenericAttributeValue<ResultMap> getResultMap();

    GenericAttributeValue<String> getScale();

    GenericAttributeValue<String> getTypeHandler();
}

File: src/main/kotlin/io/github/cdgeass/codeInsight/dom/element/Parameter.kt
// Generated on Fri Mar 26 00:14:00 CST 2021
// DTD/Schema  :    mybatis-3-mapper.dtd

package io.github.cdgeass.codeInsight.dom.element

import com.intellij.util.xml.DomElement
import com.intellij.util.xml.GenericAttributeValue
import com.intellij.util.xml.Required

/**
 * mybatis-3-mapper.dtd:parameter interface.
 * @author cdgeass
 */
interface Parameter : DomElement {

	/**
	 * Returns the value of the resultMap child.
	 * Attribute resultMap
	 * @return the value of the resultMap child.
	 */
	fun getResultMap(): GenericAttributeValue<String>

	/**
	 * Returns the value of the jdbcType child.
	 * Attribute jdbcType
	 * @return the value of the jdbcType child.
	 */
	fun getJdbcType(): GenericAttributeValue<String>

	/**
	 * Returns the value of the property child.
	 * Attribute property
	 * @return the value of the property child.
	 */
	@Required
	fun getProperty(): GenericAttributeValue<String>

	/**
	 * Returns the value of the typeHandler child.
	 * Attribute typeHandler
	 * @return the value of the typeHandler child.
	 */
	fun getTypeHandler(): GenericAttributeValue<String>

	/**
	 * Returns the value of the scale child.
	 * Attribute scale
	 * @return the value of the scale child.
	 */
	fun getScale(): GenericAttributeValue<String>

	/**
	 * Returns the value of the mode child.
	 * Attribute mode
	 * @return the value of the mode child.
	 */
	fun getMode(): GenericAttributeValue<String>

	/**
	 * Returns the value of the javaType child.
	 * Attribute javaType
	 * @return the value of the javaType child.
	 */
	fun getJavaType(): GenericAttributeValue<String>

}
